app/domain/authentication/authn_jwt/fetch_jwt_claims_to_validate.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module Authentication
+  module AuthnJwt
+
+    # FetchJwtClaimsToValidate command class is responsible to return a list of JWT standard claims to
+    # validate, according to the following logic:
+    # For each optional claim (iss, exp, nbf, iat) that exists in the token - add to mandatory list
+    # Note: the list also contains the value to validate if necessary (for example iss: cyberark.com)
+    FetchJwtClaimsToValidate ||= CommandClass.new(
+      dependencies: {
+        fetch_issuer_value: ::Authentication::AuthnJwt::FetchIssuerValue.new,
+        jwt_claim_class: ::Authentication::AuthnJwt::JwtClaim,
+        logger: Rails.logger
+      },
+      inputs: %i[authentication_parameters]
```

Comment:
just an idea - can we use the different inputs instead of putting them in the `authentication_parameters` object? one of the advantages of the command class pattern is that it is very clear what the inputs to the class are but in this case i have no idea which parameters are in `authentication_parameters`. 
wdyt?

Reply:
Because we build a generic authenticator so each implementation may need a different set of parameters for example:
GeUserIdentityFromUrl - needs only the url parameters
GetUserFromToken - needs account service id (to fetch the claim from policy) and the decoded token

So we need 1 container (authentication_parameters) in order to send the same input to both implementations 

Lets raise this challenge again in the final PR to see if we can change this idea